tests/api_connexion/endpoints/test_log_endpoint.py

```diff
@@ -14,25 +14,254 @@
+        os.makedirs(dir_path)
+        with open(f"{dir_path}/1.log", "w+") as file:
+            file.write("Log for testing.")
+            file.flush()
 
-    @pytest.mark.skip(reason="Not implemented yet")
-    def test_should_response_200(self):
+    def tearDown(self):
+        logging.config.dictConfig(DEFAULT_LOGGING_CONFIG)
+        clear_db_runs()
+
+        # Remove any new modules imported during the test run. This lets us
+        # import the same source files for more than one test.
+        for mod in [m for m in sys.modules if m not in self.old_modules]:
+            del sys.modules[mod]
+
+        sys.path.remove(self.settings_folder)
+        shutil.rmtree(self.settings_folder)
+        shutil.rmtree(self.log_dir)
+
+        super().tearDown()
+
+    @provide_session
+    def test_should_response_200_json(self, session):
+        self._create_dagrun(session)
+        key = self.app.config["SECRET_KEY"]
+        serializer = URLSafeSerializer(key)
+        token = serializer.dumps({"download_logs": False})
+        headers = {'Accept': 'application/json'}
         response = self.client.get(
-            "/dags/TEST_DG_ID/dagRuns/TEST_DAG_RUN_ID/taskInstances/TEST_TASK_ID/logs/3"
+            f"api/v1/dags/{self.DAG_ID}/dagRuns/TEST_DAG_RUN_ID/"
+            f"taskInstances/{self.TASK_ID}/logs/1?token={token}",
+            headers=headers
+        )
+        self.assertIn('content', response.json)
+        self.assertIn('continuation_token', response.json)
+        self.assertIn('Log for testing.', response.json.get('content'))
+        self.assertEqual(200, response.status_code)
+
+    @provide_session
+    def test_should_response_200_text_plain(self, session):
+        self._create_dagrun(session)
+        key = self.app.config["SECRET_KEY"]
+        serializer = URLSafeSerializer(key)
+        token = serializer.dumps({"download_logs": True})
+
+        response = self.client.get(
+            f"api/v1/dags/{self.DAG_ID}/dagRuns/TEST_DAG_RUN_ID/"
+            f"taskInstances/{self.TASK_ID}/logs/1?token={token}",
+            headers={'Accept': 'text/plain'}
         )
         assert response.status_code == 200
+        self.assertEqual(200, response.status_code)
+        self.assertIn('Log for testing.', response.data.decode('utf-8'))
+
+    @provide_session
+    def test_get_logs_response_with_ti_equal_to_none(self, session):
+        self._create_dagrun(session)
+        key = self.app.config["SECRET_KEY"]
+        serializer = URLSafeSerializer(key)
+        token = serializer.dumps({"download_logs": True})
+
+        response = self.client.get(
+            f"api/v1/dags/{self.DAG_ID}/dagRuns/TEST_DAG_RUN_ID/"
+            f"taskInstances/Invalid-Task-ID/logs/1?token={token}",
+        )
+        self.assertIn('content', response.json)
+        self.assertIn('continuation_token', response.json)
+        self.assertEqual("[*** Task instance did not exist in the DB\n]",
```

Comment:
Should we return 404 here?

Reply:
Since we are logging, I think returning it as log is cool. What do you suggest

Reply:
I think any errors should be reported as exceptions. We should not report errors otherwise, as this will make it difficult for the client to handle them.

Reply:
Fixed. Please take a look https://github.com/apache/airflow/pull/9331/commits/e858bf2fb850c8626574d7182709636f59965d77